README.md

```diff
@@ -1,15 +1,188 @@
+
+Steps:
+1. Convert raw events into it’s signature type with **parse_signature**
+2. Group events by identical signatures and compare events with different signatures Per event with **compare_events**
+3. Generate issues table by using the **report_issues** macro.
+
+
+## Installing the package
+Include the following in your `packages.yml` file in your dbt project:
+
+```
+packages:
+    - package: avo/avo_audit
+      version: ["0.1.0]
+```
+
+Run `dbt deps` in the root of your dbt project to install the package.
+
+
+# Macros
+
+### Disclaimer
+> This project is still work in progress. These macros have not been fully implemented yet, however some of the work has been started and we will be doing our best to get this up and running as soon as possible. Reach out if you'd like to help out!
+
+## parse_signature
+
+This macro generates a SQL that transforms each raw event into it’s signature.  
+Instead of using the signature from the column definitions, it is deriving the signature from the value. This can be used to compare events with identical names together to understand if and how they vary from each other.
+
+
+**Raw Events:**
+| event_name  | user_id      | user_name | device  |
+|-------------|--------------|-----------|---------|
+| App Opened  | 2c3416a3033e | John      | IOS     |
+| App Opened  | fc3416basdq1 | Donna     | IOS     |
+| App Opened  | c5e6d9429f27 | Lisa      |         |
+| App Opened  | 55331212412  | Joshua    | Android |
+
+
+
+**After running parse_signature:**
+| event_name | property_name_mapping        | property_signature_mapping |
+|------------|------------------------------|----------------------------|
+| App Opened | [user_id, user_name, device] | [string, string, string]   |
+| App Opened | [user_id, user_name, device] | [string, string, string]   |
+| App Opened | [user_id, user_name, device] | [string, string, null]     |
+| App Opened | [user_id, user_name, device] | [int, string, null]        |
+
+
+We recommend to test this in your development environment first while you get set up.
+```
+{# in dbt Develop #}
+
+{% set raw_event_relation=adapter.get_relation(
+      database=target.database,
+      schema="raw_event_schema",
+      identifier="raw_event_table"
+) -%}
+
+{{ avo_audit.parse_signature(
+    relation=raw_event_relation,
+    date_partition=”sent_at”,
+    lookback_hours=1,
+) }}
+```
+
+**Arguments:**
+* **raw_event_relation:** The database table containing your raw events which you want to compare.
+* **date_partition:** The column in raw events that can be used to filter on time to limit query.
+* **lookback_hours:** How many hours the query should look back, we recommend keeping it under 24h to keep the cost of the query low.
+
+
+
+## compare_events
+
+This macro will run on the results of parse_signature.
+
+It will execute a SQL that will go through each event_name group, and compare it together with all the other events to identify the issues documented above.
```

Comment:
```suggestion
It will go through each event_name group, and compare it together with all the other events to identify the missing properties, property type mismatches and other issues documented above.
```

Reply:
What is exactly an event_name group?